src/material/icon/trusted-types.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * @fileoverview
+ * A module to facilitate use of a Trusted Types policy internally within
+ * Angular Material. It lazily constructs the Trusted Types policy, providing
+ * helper utilities for promoting strings to Trusted Types. When Trusted Types
+ * are not available, strings are used as a fallback.
+ * @security All use of this module is security-sensitive and should go through
+ * security review.
+ */
+
+export declare interface TrustedHTML {
+  __brand__: 'TrustedHTML';
+}
+
+export declare interface TrustedTypePolicyFactory {
+  createPolicy(policyName: string, policyOptions: {
+    createHTML?: (input: string) => string,
+  }): TrustedTypePolicy;
+}
+
+export declare interface TrustedTypePolicy {
+  createHTML(input: string): TrustedHTML;
+}
+
+/**
+ * The Trusted Types policy, or null if Trusted Types are not
+ * enabled/supported, or undefined if the policy has not been created yet.
+ */
+let policy: TrustedTypePolicy|null|undefined;
+
+/**
+ * Returns the Trusted Types policy, or null if Trusted Types are not
+ * enabled/supported. The first call to this function will create the policy.
+ */
+function getPolicy(): TrustedTypePolicy|null {
+  if (policy === undefined) {
+    policy = null;
+    if (typeof window !== 'undefined') {
+      const ttWindow = window as unknown as {trustedTypes?: TrustedTypePolicyFactory};
+      if (ttWindow.trustedTypes !== undefined) {
+        policy = ttWindow.trustedTypes.createPolicy('material#icon', {
```

Comment:
I think that it would be better to create a single policy for all of @angular/material (and even @angular/cdk), but that would require us to figure out how to securely share it between all of the entry points (e.g. `@angular/material/icon`).

I don't believe that we have such sharing mechanism today. We do have a concept of private exports in other packages, but this only obfuscates the export name with a hard to type prefix — `ɵ` ([example](https://github.com/angular/angular/blob/437759ba471283cfb00d326d9c971a5fbfb8e0fe/packages/common/src/private_export.ts#L9-L10)) which is not good enough for a trusted type policy.

I wonder if there is another way to do this.

The main concern with entry-point level policy names is that there could be too many of them added over time, which would make it a nightmare to manage the policy names within the server configuration.

Reply:
aside: the name of the policy could also be changed to be more consistent with our existing angular policies, for example `angular-material` or `angular-components`.

Reply:
+1, the name of the policy should include "angular" in the name. 

As for sharing, I don't know if doing something cross-library possible? I don't know the trusted types system super well, but it seems like the policy object needs to be inaccessible to userspace, yes? If that's the case, there's no mechanism such that another material entry-point could access it while userspace couldn't. 

That said, we don't use `innerHTML` anywhere else in material or cdk- the SVG handling here for icons is special since SVGs are _assets_ that also need to be inlined into the DOM. I would like to move this svg handling code _into_ the cdk in the future, but we can cross that bridge when we get to it.

Reply:
@jelbourn what policy name would you prefer? `angular#material`, `angular#cdk`, or `angular#components`?

+1 for keeping the code here for now, as long as we don't foresee a need to do other security-sensitive DOM operations elsewhere in material or cdk.

Reply:
I changed the name to `angular#components` to allow us to use it both for material and cdk in the future. Would be happy to change this if you have other preferences.

Reply:
sgtm!